### PR TITLE
feat(launchpad): add RockRecipe

### PR DESCRIPTION
**Pull request by tigarmo**
- [ ] Have you followed the guidelines for contributing?
- [ ] Have you signed the [CLA](http://www.ubuntu.com/legal/contributors/)?
- [ ] Have you successfully run `tox`?

-----

### CODE REVIEW

File: craft_application/launchpad/models/recipe.py [hunk 1]
```diff
@@ -374,8 +376,16 @@ def new(  # noqa: PLR0913
             asset.
         :param store_channels: (Optional) The channels onto which to publish the asset
             if uploaded.
+        :param architectures: (Optional) The architectures to build. Note that this
```
Comment (tigarmo) [inline craft_application/launchpad/models/recipe.py:379]: this fix is on a separate commit

**Review by lengau — APPROVED** [review]
Love the little improvements along the way! Thanks!

File: craft_application/launchpad/models/recipe.py [hunk 2]
```diff
@@ -465,4 +475,33 @@ def _get_lp_recipe(cls, lp: Launchpad) -> Any:
...
 
 
-Recipe = SnapRecipe | CharmRecipe
+class RockRecipe(_StandardRecipe):
+    """A recipe for a rock.
+
+    https://api.launchpad.net/devel.html#rock_recipe
+    """
+
+    ARTIFACT: ClassVar[Literal["rock"]] = "rock"  # type: ignore[reportIncompatibleVariableOverride]
```
Comment (bepri) [inline craft_application/launchpad/models/recipe.py:484]: What's the benefit of declaring this as `Literal["rock"] = "rock"` over `str = "rock"`? It'd let us remove the type ignore. (Same goes for the charm one up above)

**Review by bepri — APPROVED** [review]

**Review by sergiusens — APPROVED** [review]

File: craft_application/launchpad/models/recipe.py (hunk 2 above)
Comment (tigarmo) [reply craft_application/launchpad/models/recipe.py:484]: it helps with type hinting, specially with method overloads like https://github.com/canonical/craft-application/pull/591/files#diff-ca02b0445f42e189fecddfef8e3f608b7e98ecadeef51d817e11e6c94e863ef1R129

**Pull request merged**